templates/solaris_install.sh.erb

```diff
@@ -14,5 +13,7 @@ pkgadd -a <%= @adminfile %> -d <%= @sourcefile %> -G <%= @install_options.join('
 # highly unlikely in most agent upgrade scenarios.
 
 <% @service_names.each do |service_name| %>
-start_service <%= service_name %>
+  if /opt/puppetlabs/bin/puppet resource service "${service_name:?}" | grep "ensure => 'running'" ; then
```

Comment:
So if already running, trigger a restart? That seems acceptable, although that's not what this code is doing.

Reply:
I'm also not 100% sure that the install will always leave the services running. That's why my suggestion was to do this test (and save it to a variable) before the `pkgadd` operation.

Reply:
@MikaelSmith You where right, it wouldn't start the service all the time.
I just moved the if before removing the puppet_agent package to see if the service was running.

Tested this and it works as intended.